From 69b65dc8a549b4a21f9196e9527728e02a62fb16 Mon Sep 17 00:00:00 2001
From: Chunjie Zhu <chunjie.zhu@citrix.com>
Date: Wed, 5 Mar 2014 09:16:26 -0800
Subject: [PATCH 1/3] vnc: handle guest DS with strange width
To: david.vrabel@citrix.com

Handle guest display surfaces that are not a multiple of 16 pixels in
width.

Signed-off-by: Chunjie Zhu <chunjie.zhu@citrix.com>
diff --git a/vnc.c b/vnc.c
index 6640833..ad75300 100644
--- a/vnc.c
+++ b/vnc.c
@@ -2629,7 +2629,8 @@ static int vnc_refresh_server_surface(VncDisplay *vd)
             guest_ptr  = guest_row;
             server_ptr = server_row;
 
-            for (x = 0; x < vd->guest.ds->width;
+            /* read/write 16 display pixels each iteration */
+            for (x = 0; x + 15 < vd->guest.ds->width;
                     x += 16, guest_ptr += cmp_bytes, server_ptr += cmp_bytes) {
                 if (!dirty_get_bit(&vd->guest.dirty, x / 16, y))
                     continue;
